document/js-api/index.bs

```diff
@@ -1115,6 +1147,111 @@ The algorithm <dfn>ToWebAssemblyValue</dfn>(|v|, |type|) coerces a JavaScript va
 
 </div>
 
+<h3 id="jspi">JavaScript Promise Integration</h3>
+
+A {{Suspending}} object represents a JavaScript function whose calls should be *suspended* when they return a Promise object.
+Each {{Suspending}} object has a \[[wrappedFunction]] internal slot which holds a JavaScript function.
+
+In addition, the {{promising}} function takes as argument a WebAssembly function and returns a JavaScript function that returns a Promise that is resolved when the WebAssembly function completes.
+
+<pre class="idl">
+[Exposed=*]
+partial namespace WebAssembly {
+    Function promising(Function wasmFun);
+};
+
+[LegacyNamespace=WebAssembly, Exposed=*]
+interface Suspending {
+    constructor(Function jsFun);
+};
+</pre>
+
+<div algorithm>
+    The  <dfn method for="WebAssembly">promising(|wasmFun|)</dfn> function, when invoked, performs the following steps:
+    1. If [$IsCallable$](|wasmFun|) is false, throw a {{TypeError}}.
+    1. If |wasmFun| does not have a \[[FunctionAddress]] internal slot, throw a {{TypeError}}.
+    1. Let |builder| be a new [=AbstractClosure=] with parameters that captures |wasmFun| that, when invoked, performs [=run a Promising function=].
+    1. Returns [=CreateBuiltinFunction=](|builder|,1,"",<<>>) 
+</div>
+
+<div algorithm>
+  The algorithm to <dfn>run a Promising function</dfn> from the JavaScript object |wasmFun| and a [=list=] of [=WebAssembly values=] |arguments| consists of the following steps:
+    1. Let |promise| be a new [=PromiseCapabilityRecord=].
+    1. Let |funcaddr| be the value of |wasmFun|'s \[[FunctionAddress]] internal slot.
+    1. Let |runner| be a new [=AbstractClosure=] of no arguments that captures |promise|, |funcaddr|, |arguments| that, when invoked, performs the following steps:
+        1. Let |functype| be [=func_type=](|store|, |funcaddr|).
+        1. Let |args| be the result of [=coerce JavaScript arguments|coercing arguments=] (|functype|,|arguments|)
+        1. Let (|store|, |result|) be the result of [=func_invoke=](|store|, |funcaddr|, |args|).
+        1. Assert: if control reaches here, we have done waiting for suspended imports
+        1. If |result|.\[[Type]] is NORMAL,
+            1. Perform [=EvaluateCall=] (|promise|.\[[Resolve]],|result|.\[[Value]],false)
+            1. Return UNDETERMINED
+        1. If |result|.\[[Type]] is THROW,
+            1. Perform [=EvaluateCall=] (|promise|.\[[Reject]],|result|.\[[Value]],false).
+            1. Return UNDETERMINED
+        1. Assert fatal error
+    1. Let |con| be [=CreateBuiltinFunction=](|runner|,3,"",[])
```

Comment:
Why is the arity 3 if `|runner|` takes no arguments?

Also, you previously used `<<>>` instead of `[]` as the last argument to `CreateBuiltinFunction`. Which one is correct? The document elsewhere uses `« »` to create lists.